lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -248,6 +261,21 @@ public function forAll(Closure $p) : bool;
      */
     public function map(Closure $func) : self;
 
+    /**
+     * Applies iteratively the given function to each element in the collection,
+     * so as to reduce the collection to a single value.
+     *
+     * @param mixed $initial
+     *
+     * @return mixed
+     *
+     * @psalm-template U
+     * @psalm-param Closure(U=,T=):U $func
+     * @psalm-param U $initial
+     * @psalm-return U
```

Comment:
This is not accurate. If Collection is empty, array_reduce will return `null`, no matter what initial value you specify.

Also, no matter what initial value you specify, closure can return some completely different data type than initial value.

This is more like it https://github.com/vimeo/psalm/issues/846

Reply:
Done